src/test/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupportSpec.scala

```diff
@@ -17,102 +20,214 @@ class DataUseRestrictionSupportSpec extends FreeSpec with Matchers with DataUseR
-          val durs: Map[String, DataUseRestriction] = booleanDatasets.flatMap { ds =>
-            Map(ds.name -> makeDurFromWorkspace(ds))
-          }.toMap
-
-          booleanCodes.map { code =>
-            val dur: DataUseRestriction = durs(code)
-            checkBooleanTrue(dur, code) should be(true)
-          }
+        "and consent codes are a mixture of true and false" in {
+          val ontologyDAO = new MockOntologyDAO
+          val request = StructuredDataRequest(generalResearchUse = false,
+            healthMedicalBiomedicalUseOnly = true,
+            diseaseUseOnly = Array("1240"),
+            commercialUseProhibited = false,
+            forProfitUseProhibited = true,
+            methodsResearchProhibited = false,
+            aggregateLevelDataProhibited = false,
+            controlsUseProhibited = true,
+            genderUseOnly = "male",
+            pediatricResearchOnly = false,
+            irbRequired = true,
+            prefix = Some("library"))
+
+          val expected = Map("libraryconsentCodes" -> Array("NPU","RS-G","RS-M","HMB","NCTRL","IRB","DS:leukemia").toJson,
+            "librarydulvn" -> FireCloudConfig.Duos.dulvn.toJson,
+            "librarystructuredUseRestriction" -> Map(
+              "NPU" -> true.toJson,
+              "RS-PD" -> false.toJson,
+              "NCU" -> false.toJson,
+              "RS-G" -> true.toJson,
+              "IRB" -> true.toJson,
+              "NAGR" -> false.toJson,
+              "RS-FM" -> false.toJson,
+              "RS-M" -> true.toJson,
+              "NMDS"-> false.toJson,
+              "NCTRL" -> true.toJson,
+              "GRU" -> false.toJson,
+              "HMB" -> true.toJson,
+              "DS" -> Array("1240").toJson).toJson)
+
+          val result = generateStructuredUseRestrictionAttribute(request, ontologyDAO)
+          result should be (expected)
         }
+      }
 
-        "dataset should have the correct list values for the consent code for which it was specified" in {
-          val durs: Map[String, DataUseRestriction] = listDatasets.flatMap { ds =>
-            Map(ds.name -> makeDurFromWorkspace(ds))
-          }.toMap
-
-          listCodes.foreach { code =>
-            val dur: DataUseRestriction = durs(code)
-            checkListValues(dur, code)
-          }
-        }
+      "when there are library data use restriction fields" - {
 
-        "dataset should have the correct disease values for the consent code for which it was specified" in {
-          val durs: Map[String, DataUseRestriction] = diseaseDatasets.flatMap { ds =>
-            Map(ds.name -> makeDurFromWorkspace(ds))
-          }.toMap
-
-          Seq("DS").foreach { code =>
-            val dur: DataUseRestriction = durs(code)
-            checkDiseaseValues(dur, code)
-          }
-        }
+//        "dataset should have a fully populated data use restriction attribute" in {
```

Comment:
Lots of test cases commented out here that I'm thinking shouldn't be:
* dataset should have a fully populated data use restriction attribute
* dur should have appropriate gender codes populated
* dur should have appropriate NAGR code populated
* dataset should have a true value for the consent code for which it was specified
* dataset should have the correct list values for the consent code for which it was specified
* dataset should have the correct disease values for the consent code for which it was specified
* dataset should not have any data use restriction for empty attributes
* dataset should not have any data use restriction for non-library attributes
* valid datasets should have some form of data use display attribute
* datasets with single boolean code should have that single display code